code/modules/clothing/head/misc.dm

```diff
@@ -28,6 +28,28 @@
 	item_state = "that"
 	flags = FPRINT
 	siemens_coefficient = 0.9
+	var/magichatchance = 5 //%
+	var/ismagichat
+	var/timer
+
+/obj/item/clothing/head/that/New()
+	if(prob(magichatchance))
+		desc = "It's a magic looking hat."
+		ismagichat = TRUE
+		wizard_garb = TRUE
+
+/obj/item/clothing/head/that/attackby(obj/item/W, mob/user)
+	if(ismagichat)
```

Comment:
might as well just check for if it's wizard_garb

Reply:
I guess I can set it to various %s.